hummingbot/data_feed/candles_feed/candles_base.py

```diff
@@ -207,14 +211,19 @@ def _reset_candles(self):
     async def fetch_candles(self,
                             start_time: Optional[int] = None,
                             end_time: Optional[int] = None):
-        rest_assistant = await self._api_factory.get_rest_assistant()
+        if start_time is None and end_time is None:
+            raise ValueError("Either the start time or end time must be specified.")
+
+        missing_records = self._candles.maxlen - len(self._candles)
+        candles_to_fetch = min(self.candles_max_result_per_rest_request - 1, missing_records)
         if end_time is None:
-            end_time = start_time + self.interval_in_seconds * self.max_records
-        # TODO: @drupman review this logic, binance spot wasn't working with this line.
-        # if start_time is None:
-        #     start_time = end_time - self.interval_in_seconds * self.max_records
+            end_time = start_time + self.interval_in_seconds * candles_to_fetch
+        if start_time is None:
```

Comment:
I removed this since the collection of binance candles is not working, I think that the API prioritizes the start time and when collecting the historical data is doing it from the end time up to the start time